From 353898a3b963f1ae8e28ce5e57190b0ffbe64340 Mon Sep 17 00:00:00 2001
From: Colin Ian King <colin.king@canonical.com>
Date: Tue, 2 Dec 2014 11:33:16 +0000
Subject: [PATCH 08/10] Improve accuracy of unit_value calculation

Coverity Scan reports:

integer_division: Dividing integer expressions 100 and
this->max_state, and then converting the integer quotient
to type float. Any remainder, or fractional part of the
quotient, is ignored.

So divide 100.0 by max_state rather than integer 100 for
improved accuracy.

Signed-off-by: Colin Ian King <colin.king@canonical.com>
---
 src/thd_cdev_intel_pstate_driver.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/thd_cdev_intel_pstate_driver.cpp b/src/thd_cdev_intel_pstate_driver.cpp
index 8750a73..775d656 100644
--- a/src/thd_cdev_intel_pstate_driver.cpp
+++ b/src/thd_cdev_intel_pstate_driver.cpp
@@ -98,7 +98,7 @@ int cthd_intel_p_state_cdev::update() {
 	thd_log_info("Use Default pstate drv settings\n");
 	max_state = default_max_state;
 	min_compensation = 0;
-	unit_value = 100 / max_state;
+	unit_value = 100.0 / max_state;
 	curr_state = 0;
 	thd_log_debug(
 			"cooling dev index:%d, curr_state:%d, max_state:%d, unit:%f, min_com:%d, type:%s\n",
-- 
2.1.3

